fn/fn.go

```diff
@@ -0,0 +1,102 @@
+package fn
+
+// Unit is a type alias for the empty struct to make it a bit less noisy to
+// communicate the informationaless type.
+type Unit = struct{}
+
+// Comp is left to right function composition. Comp(f, g)(x) == g(f(x)). This
+// can make it easier to create on the fly closures that we may use as
+// arguments to other functions defined in this package (or otherwise).
+func Comp[A, B, C any](f func(A) B, g func(B) C) func(A) C {
+	return func(a A) C {
+		return g(f(a))
+	}
+}
+
+// Iden is the lef and right identity of Comp. It is a function that simply
```

Comment:
```suggestion
// Iden is the left and right identity of Comp. It is a function that simply
```

Reply:
And would be nice to just name it `Identity`.

Reply:
We can do so but I'd like to offer a couple of reasons I chose `Iden` instead.

1. We have an 80 column limit. This space is quite scarce and `Iden` takes 5% of it whereas `Identity` would take 10%.
2. Simplicity opts for the names [`iden` and `comp`](https://github.com/BlockstreamResearch/simplicity/blob/master/Haskell/Core/Simplicity/Term/Core.hs#L42-L52) instead of `identity` and `compose`. Given that it is a well-known Bitcoin project I figured it would provide good intuition